Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,344 @@
+            };
+
+            // Lock and limit motion axes
+            _configurableJoint.xMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.yMotion = ConfigurableJointMotion.Limited;
+            _configurableJoint.zMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularXMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularYMotion = ConfigurableJointMotion.Locked;
+            _configurableJoint.angularZMotion = ConfigurableJointMotion.Locked;
+
+            // Adjust anchor position for button travel distance
+            _configurableJoint.anchor = new Vector3(
+                _configurableJoint.anchor.x,
+                _configurableJoint.anchor.y - (_buttonTravelDistance / 2),
+                _configurableJoint.anchor.z);
+
+            _configurableJoint.connectedAnchor = new Vector3(
+                _configurableJoint.connectedAnchor.x,
+                _configurableJoint.connectedAnchor.y - (_buttonTravelDistance / 2),
+                _configurableJoint.connectedAnchor.z);
+
+            // Set linear limit for button travel
+            _configurableJoint.linearLimit = new SoftJointLimit
+            {
+                limit = _buttonTravelDistance
+            };
+
+            _configurableJoint.targetPosition = _initialButtonPosition;
+        }
+
+        private void FixedUpdate()
+        {
+            if(_rigidbody.velocity.sqrMagnitude > _buttonVelocityThreshold.sqrMagnitude)
+            {
+                FreezePressableMovement();
+            }
+            else
+            {
+                UnFreezePressableMovement();
+            }
+
+            float distance = Mathf.Abs(_pressableObject.transform.localPosition.y - _initialButtonPosition.y);
+
+            // Check if the button should be pressed
+            if (!_isButtonPressed && distance >= _buttonTravelDistance &&
+                (_contactHandPressing || (!_shouldOnlyBePressedByHand && _objectsContactingButton.Count > 0)))
+            {
+                _isButtonPressed = true;
+                ButtonPressed();
+            }
+
+            // Check if the button should be released
+            if (_isButtonPressed && distance < _buttonTravelDistance * BUTTON_PRESS_EXIT_THRESHOLD)
+            {
+                _isButtonPressed = false;
+                ButtonUnpressed();
+            }
+        }
+
+        /// <summary>
+        /// Invoke the button pressed event.
+        /// </summary>
+        protected virtual void ButtonPressed()
+        {
+            OnButtonPressed?.Invoke();
+        }
+
+        /// <summary>
+        /// Invoke the button released event.
+        /// </summary>
+        protected virtual void ButtonUnpressed()
+        {
+            OnButtonUnPressed?.Invoke();
+        }
+
+        /// <summary>
+        /// Handle hand contact with the pressable object.
+        /// </summary>
+        public void OnHandContactPO(ContactHand hand)
```

Comment:
Could these callback functions be virtual, to allow the Toggle (or other extended classes) to just override them?